node.js/week2/homework/meal-sharing/src/backend/api/meals-router.js

```diff
@@ -1,12 +1,61 @@
+const { request, response } = require("express");
 const express = require("express");
 const router = express.Router();
 
 const meals = require("./../data/meals.json");
 
 router.get("/", async (request, response) => {
   try {
-    console.log(meals);
-    console.log("in /api/meals");
+    const maxPrice = request.query.maxPrice
+    const title = request.query.title
+    const createdAfter = request.query.createdAfter
+    const limit = request.query.limit
+    if (maxPrice) {
+      if (!Number.isInteger(parseInt(maxPrice))) {
+        response.statusCode = 400;
+        response.json({ "message": "bad input" });
+        return;
+      } else {
+        response.send(meals.filter(meal => meal.price <= maxPrice))
+      }
+    } else if (title) {
+      response.send(meals.filter(meal => meal.title.includes(title)))
+    } else if (createdAfter) {
+      if (!Date.parse(createdAfter)) {
+        response.statusCode = 400;
+        response.json({ "message": "bad input" });
+        return;
+      } else {
+        response.send(meals.filter(meal => Date.parse(meal.createdAt) >= Date.parse(createdAfter)))
+      }
+    } else if (limit) {
+      if (!Number.isInteger(parseInt(limit))) {
+        response.statusCode = 400;
+        response.json({ "message": "bad input" });
+        return;
+      } else {
+        response.send(meals.slice(0, limit))
+      }
+    } else {
+      response.send(meals);
+    }
+  } catch (error) {
+    throw error
+  }
+})
+
+router.get("/:id", async (request, response) => {
+  try {
+    const id = request.params.id
+    if (id > meals.length) {
```

Comment:
Id does necessarily not start with 0 or 1.
What if id's start with 101, meals.length is 20, and id param given is 105?
So you don't need this logic. (loop in line 50).